### ISSUE TITLE
Handle relative paths in `-r` links

### ISSUE THREAD

**Issue by malikoth** [di/pip-api#35]
This project is awesome, thank you so much!

OK, for context, here's the structure of requirements that I have:

```
requirements.txt
reqs/base.txt
reqs/dev.txt
```

`requirements.txt` contains the following:

```
-r reqs/dev.txt
```

`reqs/dev.txt` contains the following:

```
-r base.txt
werkzeug
```

`reqs/base.txt` contains the following:

```
django
```

This is a completely valid structure, as I can run `pip install -r requirements.txt` and I get both django and werkzeug installed.  If I modify the first line of `reqs/dev.txt` to read `-r reqs/base.txt`, then I get the following error upon running `pip install -r requirements.txt`: `ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'reqs/reqs/base.txt'`  So pip requires that references to other requirements files be expressed as a path relative to the file being parsed.

OK, so here's the problem.  I call `pip_api.parse_requirements('requirements.txt')`, and I get the following (excerpt of a) traceback:

```
    reqs = pip_api.parse_requirements('requirements.txt')
  File "/usr/local/lib/python3.7/site-packages/pip_api/_parse_requirements.py", line 183, in parse_requirements
    lines = "".join(_read_file(filename)).replace("\\\n", "").splitlines()
  File "/usr/local/lib/python3.7/site-packages/pip_api/_parse_requirements.py", line 47, in _read_file
    with open(filename) as f:
FileNotFoundError: [Errno 2] No such file or directory: 'base.txt'
```

So it's reading `requirements.txt`, finding the line pointing to `reqs/dev.txt`, parsing it, finding a reference to `base.txt`, and then failing because it can't find that file in the repository root next to `requirements.txt`.  

`pip_api` needs to handle file paths as being relative to the file being parsed as pip does, because setting them to be parseable by `pip_api` right now makes it so that my `pip install` command no longer works.  If you agree with my conclusion, I'd be happy to submit a PR with a fix.

**Comment by di** [comment]
> This project is awesome, thank you so much!

You're very welcome! Sounds like you've uncovered a valid bug.

> I'd be happy to submit a PR with a fix.

Please do, and please make sure to include a failing/fixed test as well which demonstrates this bug.

One note: CI for this project takes a really long time to run, since we're testing `N`x`M` combinations of `pip` version and Python version. However, I doubt this bug behaves differently in a different version of `pip` or Python, so I'd recommend getting everything passing locally with just `tox -e py37-pip1911` and _then_ submit a PR which runs the whole suite once.

**Comment by malikoth** [comment]
> a failing/fixed test

A ... a what?  This sentence makes a whole lot more sense to me without either of those f-words in it.  The test I wrote started out failing, and now it's fixed, is that what you mean?

**Comment by di** [comment]
Yep 🙂

**Comment by di** [comment]
Released in [`pip-api==0.0.11`](https://pypi.org/project/pip-api/0.0.11/).
